src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -90,6 +90,53 @@ class Wavefunction
 
     void flush() const
     {
+        //logic to reorder
+        Fusion fg = fused_.getFusedGates();
+        auto items = fg.getItems();
+        auto sets = fg.getSet();
+        auto ctrlSets = fg.getCtrlSet();
+        //getting all qubits to move to lower end of the wfn
+        if (items.size() > 0) {
+            std::vector<unsigned> indices;
+            std::set<unsigned> indicesSet;
+            for (int i = 0; i < items.size(); i++) {
+                auto tempIndices = items[i].get_indices();
+                for (unsigned j = 0; j < tempIndices.size(); j++) {
+                    if (indicesSet.count(tempIndices[j]) == 0) {
+                        indices.push_back(tempIndices[j]);
+                        indicesSet.insert(tempIndices[j]);
+                    }
+                }
+            }
+            for (auto it = ctrlSets.begin(); it != ctrlSets.end(); ++it) {
+                if (indicesSet.count(*it) == 0) {
+                    indices.push_back(*it);
+                    indicesSet.insert(*it);
+                }
+            }
+            //performing reorder
+            std::vector<qubit_t> indexLocs = qubits(indices);
```

Comment:
This seems rather inefficient: the qubits vector gets copied as a whole after each transposition of two qubits. Maybe, refactor to deal with in-place transposition of qubitmap here and use (change it to private) reorderWFN to update wfn_'s state only?

Reply:
I will hold this of until the efficient reorder routine is implemented over the next week or so!